### PR TITLE
Refactor compiler build flags to one location

### PR DESCRIPTION
**Pull request by kgpai**
Currently Velox build flags are sprinkled across not only our code base , but several dependent code bases. Since Velox is targeting multiple cpu architectures, it makes sense to have one single source to determine the appropriate flags to use through out this and dependent projects. 
This pr moves this logic to a shell function  so we can get the right flags for the setup scripts, and refactors scripts to call this.

### CODE REVIEW

File: scripts/setup-helper-functions.sh [hunk 1]
```diff
@@ -37,6 +37,84 @@ function github_checkout {
...
+            CPU_ARCH="avx"
+      elif [[ $CPU_CAPABILITIES =~ "sse" ]]; then
+            CPU_ARCH="sse"
+      elif [ $MACHINE = "aarch64" ]; then
+            CPU_ARCH="aarch64"
+      fi
+    fi
+  fi
+
+  case $CPU_ARCH in
```
Comment (majetideepak) [inline scripts/setup-helper-functions.sh:97]: if the input `CPU_ARCH` is invalid or incorrectly typed, should we throw a warning message that an invalid option was specified?

File: CMakeLists.txt [hunk 2]
```diff
@@ -151,12 +151,14 @@ set(POSITION_INDEPENDENT_CODE True)
...
-  set(CMAKE_CXX_FLAGS
-      "${CMAKE_CXX_FLAGS} -mavx2 -mfma -mavx -mf16c -mbmi2 -mbmi")
-endif()
+
+execute_process(
+  COMMAND
+    bash -c
+    "( source ./scripts/setup-helper-functions.sh; echo -n $(get_cxx_flags))"
+  OUTPUT_VARIABLE SCRIPT_CXX_FLAGS)
+
```
Comment (majetideepak) [inline CMakeLists.txt:160]: A status message here to show SCRIPT_CXX_FLAGS (what architecture-specific flags are being used) will be helpful to debug

Comment (kgpai) [reply CMakeLists.txt:160]: FYI, You can run in --trace-source or --trace-expand mode to get variable values when debugging etc. But I will add this as you suggest.

Comment (majetideepak) [reply CMakeLists.txt:161]: We can avoid it if the output is verbose. I am not very particular if `--trace-source` does the job.

**Pull request closed without merging**